tests/integration/test_diagnostic_run.py

```diff
@@ -50,63 +49,73 @@ def check(result_file):
     assert not missing
 
 
-SCRIPTS = {
-    'diagnostic.py':
-    dedent("""
-        import yaml
-        from esmvaltool.diag_scripts.shared import run_diagnostic
-
-        def main(cfg):
-            with open(cfg['setting_name'], 'w') as file:
-                yaml.safe_dump(cfg, file)
-
-        if __name__ == '__main__':
-            with run_diagnostic() as config:
-                main(config)
-        """),
-    'diagnostic.ncl':
-    dedent("""
-        begin
-            print("INFO    Loading settings from " + getenv("settings"))
-            loadscript("$settings")
-        end
-        print("INFO Writing " + diag_script_info@setting_name)
-        n = str_get_nl()
-        result = "run_dir: " + config_user_info@run_dir + n +\
-                 "work_dir: " + config_user_info@work_dir + n +\
-                 "plot_dir: " + config_user_info@plot_dir + n +\
-                 "log_level: " + config_user_info@log_level + n +\
-                 "input_files: []" + n
-
-        system("echo '" + result + "' > " + diag_script_info@setting_name)
-        """),
-    'diagnostic.R':
-    dedent("""
-        library(yaml)
-
-        args <- commandArgs(trailingOnly = TRUE)
-        print(paste0("INFO    Loading settings from ", args[1]))
-        settings <- yaml::read_yaml(args[1])
-
-        print(paste0("INFO    Writing settings to ", settings$setting_name))
-        yaml::write_yaml(settings, settings$setting_name)
-        """),
-    'diagnostic.jl':
-    dedent("""
-        import YAML
-        @info "Starting diagnostic script with" ARGS
-        config_file = ARGS[1]
-        cfg = YAML.load_file(config_file)
-        out_file = cfg["setting_name"]
-        @info "Copying file to" out_file
-        Base.Filesystem.cp(config_file, out_file)
-        @info "Done"
-    """),
-}
+RECIPE_PY = """
```

Comment:
If you're going to change this anyway, It would be nicer to store these diagnostics scripts in separate files (diagnostic.py, diagnostic.R, etc), because then syntax highlighting works properly and they can be linted.